File: 3003/3003.c
int maximizeTheNumberOfPartitionsA(struct TreeNode* root) {
    if (!root) return 0;
    int l = maximizeTheNumberOfPartitionsA(root->left);
    int r = maximizeTheNumberOfPartitionsA(root->right);
    return 1 + (l > r ? l : r);
}

File: 3003/3003.cs
public class Solution {
    public int MaximizeTheNumberOfPartitionsA(TreeNode root) {
        if (root == null) return 0;
        return 1 + Math.Max(MaximizeTheNumberOfPartitionsA(root.left), MaximizeTheNumberOfPartitionsA(root.right));
    }
}